### ISSUE TITLE
Vehicle in traffic, even though other vehicle has moved out of the way

### ISSUE THREAD

**Issue by sootysax** [Courseplay/courseplay#2177]
I think this issue was raised and closed before..  but it is happening to me..   I passed in front of a tractor that was on an unload run with my bale trailer, and even though i went past him pretty quickly, the CP vehicle remained 'in traffic' until i stopped and started the course again.

**Comment by dkfiremann** [comment]
Just to help out, as I raised the last issue: 
What version of CP are you using? 
A log with debug 3 enabled will have de details on traffic.

**Comment by sootysax** [comment]
i will try and recreate, do you just need a log with debug 3 or screenshot with collisions visible?

**Comment by andreasthu** [comment]
i can repeat this
vehicle in traffic, the the tractor stops while an ai-car/traffic is in the way,
when the car, which block my vehicle, is away the tractor still stand and dont drive away itself.
onyl fire and hire helps
version 01.0164dev
in earlyer version this was not so
lg
andreas

**Comment by andreasthu** [comment]
here is the log:
[log.txt](https://github.com/Courseplay/courseplay/files/1141050/log.txt)
and two pics
![1](https://user-images.githubusercontent.com/24860501/28104125-cad31b9a-66d9-11e7-9717-0340756e05e0.png)
![2](https://user-images.githubusercontent.com/24860501/28104127-cf9eb58a-66d9-11e7-8f78-e2d2ddae59ac.png)

hope it helps
lg
andrea

**Comment by dkfiremann** [comment]
It will be good if you activate the debug channel before starting the course. 
Form the log, it is not visible what objects are on the ignore list for the vehicle. 
That is only logged when you start the course.

**Comment by pops64** [comment]
Yes like dkfiremann said debug channel 3 needs to be on before encountering the issue. This holds true whenever you use a debug channel.

**Comment by sootysax** [comment]
this is still happening...  though it seems only when the savegame has been running for a while..  i will try and capture this behavior with a channel 3 running.. but when it happens, its ANY time another of my vehicles gets near a CP course vehicle.. then i have to stop and restart course

**Comment by dkfiremann** [comment]
What version are you using? 
a lot of changes has happend since .0165

**Comment by MoslessNeo** [comment]
In the version I'm using, 5.01.00177, several fixes were made, mostly for the DriveControl.

**Comment by pops64** [comment]
I think this is occurring because the when the fieldwork helper is in the turn maneuver it isn't clearing the traffic collision.

**Comment by pops64** [comment]
[Log](https://pastebin.com/6VVMKfPg) with channel three active using above bug to reproduce.

**Comment by pops64** [comment]
Alright I was reviewing my log and I think i found the hic up 
```
N Series (#3): 	self.cp.collidingObjects[4]:
N Series (#3): 	                             55206 agco1100
N Series (#3): 	                             53124 stollFZ
N Series (#3): 	                             52792 kuhnVariMaster153
```
53124 was on the vehicle but then removed but some how it never got deleted from self.cp.colldingObject[4]

**Comment by pops64** [comment]
Posting this here as the most reecent version seems to be detecting things that aren't even collisions or there or remembering  past collisions. https://pastebin.com/vefe4MSS

**Comment by Zoochase** [comment]
I am still seeing this with the released version as well.

**Comment by pops64** [comment]
This is a very rare bug to come across and has been hard to fix.

**Comment by pops64** [comment]
Closed #2909
